### DISCUSSION TITLE
Currency format

### DISCUSSION THREAD

**Discussion by paulustimothy** [filamentphp/filament#12489] [Help] [answered]
### Package

Table builder

### Package Version

3.2.71

### How can we help you?

Is there a way to remove or hide decimal part in table builder, for example 10.000,00 to be 10.000

my code

`TextColumn::make('price')
                   ->sortable()
                    ->money('IDR', locale: 'id'),`
thanks

**Comment by leandrocfe** [comment]
[formatStateUsing](https://filamentphp.com/docs/3.x/tables/columns/text#custom-formatting)?

**Reply by paulustimothy** [reply]
im new to filament, can you tell me how to use formatStateUsing for this problem?

**Reply by leandrocfe** [reply, marked as answer]
It could be better, but I think it works

```php
->formatStateUsing(fn ($state) => Str::of(Number::currency($state, 'IDR', 'id'))->replace(',00', ''))
```

**Comment by rachmathikam** [comment]
TextInput::make('nominal')
                                ->label('Jumlah (IDR)')
                                ->required()
                                ->reactive()
                                ->extraAttributes([
                                    'id' => 'rupiah',
                                    'oninput' => 'formatRupiah(this)',
                                ])
                                ->afterStateHydrated(function ($state, callable $set) {
                                    $set('nominal', 'Rp. ' . number_format($state, 0, ',', '.'));
                                })
                                ->dehydrateStateUsing(fn ($state) => str_replace(['Rp. ', '.'], '', $state))
                                ->prefix('Rp. ')
                                ->hint('Masukkan jumlah uang dalam Rupiah'),
                                

How do I input Indonesian currency when typing in the form

**Reply by leandrocfe** [reply]
Use [input-masking](https://filamentphp.com/docs/3.x/forms/fields/text-input#input-masking)
